### PR TITLE
feat: Allow tracing to get log and session id

### PR DESCRIPTION
**Pull request by kaavee315**
> [!IMPORTANT]
> Add `allow_tracing` parameter to enable tracing in Composio SDK functions and update tests accordingly.
> 
>   - **Behavior**:
>     - Add `allow_tracing` parameter to `_execute()` in `__init__.py`, `execute()` in `collections.py`, and `execute_action()` in `toolset.py`.
>     - Pass `allow_tracing` through function calls to enable tracing functionality.
>   - **Tests**:
>     - Update `test_execute_action_param_serialization()` in `test_toolset.py` to include `allow_tracing` parameter.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for 6fd429807868936ff0084b1f627ba3e77098e894. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: python/composio/tools/toolset.py [hunk 1]
```diff
@@ -1454,6 +1454,7 @@ def __init__(
         output_dir: t.Optional[Path] = None,
         output_in_file: bool = False,
         verbosity_level: t.Optional[int] = None,
+        allow_tracing: bool = False,
```
Comment (shreysingla11) [inline python/composio/tools/toolset.py:1457]: Consider adding validation for when tracing is enabled but session_id is None, as tracing information might be incomplete without a session context. You could add a warning log in such cases.

File: python/composio/client/collections.py [hunk 2]
```diff
@@ -1329,6 +1330,7 @@ def execute(
                         "sessionInfo": {
                             "sessionId": session_id,
                         },
+                        "allowTracing": allow_tracing,
```
Comment (shreysingla11) [inline python/composio/client/collections.py:1333]: Consider defining `allowTracing` as a constant at the module level to avoid string literals in the code. This would make maintenance easier and prevent typos.

**Comment by shreysingla11** [comment]
### Code Review Summary

#### Overall Assessment
The changes to add tracing functionality are generally well-structured and follow the existing patterns in the codebase. The implementation is consistent across all affected files and maintains type safety.

#### Strengths
- Consistent implementation across the execution chain
- Good use of type hints
- Proper default values (False) making it an opt-in feature
- Clean integration with existing session management

#### Areas for Improvement
1. Documentation
   - Missing docstring updates for new parameters
   - Could benefit from example usage in docstrings

2. Code Structure
   - String literals used for API parameters could be constants
   - Potential edge case with tracing enabled but no session_id

3. Testing
   - No test cases included for the new functionality
   - Should add tests covering tracing enabled/disabled scenarios

#### Recommendations
1. Add parameter documentation in docstrings
2. Add validation/warning for tracing without session_id
3. Define constants for API parameter names
4. Add unit tests for the new functionality

#### Rating: 7/10
Good implementation but needs documentation improvements and tests.

**Review by plxity — APPROVED** [review]

**Review by angrybayblade — APPROVED** [review]
